web/modules/custom/dpl_fbi/src/Plugin/Field/FieldWidget/WorkIdSearchForMaterialWidget.php

```diff
@@ -0,0 +1,108 @@
+ * )
+ */
+final class WorkIdSearchForMaterialWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state): array {
+
+    $uniqId = uniqid((string) rand(), TRUE);
+    $identifier = "{$uniqId}-{$delta}";
+
+    $element['fields_container'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'class' => ['material-search__inputs-container--hidden'],
+      ],
+    ];
+
+    $element['fields_container']['value'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Field work ID'),
+      '#default_value' => $items[$delta]->value ?? '',
+      '#attributes' => [
+        'data-field-input-work-id' => $identifier,
+      ],
+    ];
+
+    $element['fields_container']['material_type'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Field material type'),
+      '#default_value' => $items[$delta]->material_type ?? '',
+      '#attributes' => [
+        'data-field-input-material-type-id' => $identifier,
+      ],
+    ];
+
+    $element['react_app_container'] = [
+      '#type' => 'container',
+      '#theme' => 'dpl_react_app',
+      '#name' => 'material-search',
+      '#data' => [
+        'class' => ['react-app-container'],
+        'previously-selected-work-id' => $items[$delta]->value ?? NULL,
+        'previously-selected-material-type' => $items[$delta]->material_type ?? NULL,
+        'unique-identifier' => $identifier,
+        'material-search-search-input-text' => $this->t('Search for material', [], ['context' => 'material search']),
+        'material-search-material-type-selector-text' => $this->t('Select material type', [], ['context' => 'material search']),
+        'material-search-material-type-selector-none-option-text' => $this->t('Select material type', [], ['context' => 'material search']),
+        'material-search-no-material-selected-text' => $this->t('No material selected', [], ['context' => 'material search']),
+        'material-search-preview-title-text' => $this->t('Title', [], ['context' => 'material search']),
+        'material-search-preview-author-text' => $this->t('Author', [], ['context' => 'material search']),
+        'material-search-preview-publication-year-text' => $this->t('Publication year', [], ['context' => 'material search']),
+        'material-search-preview-source-text' => $this->t('Source', [], ['context' => 'material search']),
+        'material-search-preview-work-id-text' => $this->t('Work ID', [], ['context' => 'material search']),
+        'material-search-loading-text' => $this->t('Loading...', [], ['context' => 'material search']),
+        'material-search-amount-of-results-text' => $this->t('Total amount of results', [], ['context' => 'material search']),
+        'material-search-aria-button-select-work-with-text' => $this->t('Select work with', [], ['context' => 'material search']),
```

Comment:
Remember to change to `Select work with title @title` if you integrate placeholders within your translation in React.